entity.go

```diff
@@ -35,10 +35,18 @@ type Entity[T any] interface {
 // NewEntity instantiates a new T with the given events applied to it.
 // At the same time the entity's internal Stream is initialised,
 // so it can record new upcoming stream.
-func NewEntity[T Entity[T]](id string, eventsSlice []VersionedEvent[T]) (*T, error) {
+func NewEntity[T Entity[T]](
```

Comment:
We could consider keeping this one and adding `NewEntityWithStreamType` for easier use when we don't care about the stream type (like in the in-memory repository).

Reply:
https://github.com/ThreeDotsLabs/esja/pull/32/commits/b7c117c49f7744700f7e26c40cb20a33d748b23f